langchain4j-alloydb-pg/src/main/java/dev/langchain4j/store/embedding/alloydb/AlloyDBEmbeddingStore.java

```diff
@@ -381,6 +384,61 @@ public void addAll(List<String> ids, List<Embedding> embeddings, List<TextSegmen
         }
     }
 
+    /**
+     * Create index in the vector store table
+     * @param index, index to be applied
+     * @param name, name of the index
+     * @param concurrently, CONCURRENTLY option
+     */
+    public void applyVectorIndex(BaseIndex index, String name, Boolean concurrently) {
+        String function;
+        if (index == null) throw new IllegalStateException("Cannot apply provided index because it is null.");
+
+        try (Connection conn = engine.getConnection(); ) {
+            if (index instanceof ScaNNIndex scaNNIndex) {
+                conn.createStatement().executeQuery("CREATE EXTENSION IF NOT EXISTS alloydb_scann");
+                function = scaNNIndex.getDistanceStrategy().getScannIndexFunction();
+            } else {
+                function = index.getDistanceStrategy().getIndexFunction();
+            }
+
+            String filter = (index.getPartialIndexes() != null
+                            && index.getPartialIndexes().isEmpty())
```

Comment:
nit: we could simplify to 
```suggestion
            String filter = isNotNullOrEmpty(index.getPartialIndexes())
```

Reply:
@averikitsch , it will throw a NullPointerException when index parameter is null

Reply:
just saw the comment above...

Reply:
oh actually partial indexes are in a List, so `isNotNullOrEmpty` cannot be called

Reply:
Oops, thanks!